### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by timja**
The tests in this project don't even compile...

cc @NotMyFault 

also cc @dduportal seems my concern was valid? =/

https://github.com/jenkins-infra/account-app/pull/137#issuecomment-1217969828

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+pipeline {
+    agent {
+        label: 'jdk11'
```
Comment (timja) [inline Jenkinsfile:3]:
```suggestion
        label 'jdk11'
```

**Comment by NotMyFault** [comment]
> The tests in this project don't even compile...

Yes, it didn't work before and doesn't work now. There was no CI before to build the actual application.

**Comment by timja** [comment]
Application doesn't even start though with your recent changes 👅.

I have fixed locally, will push soon

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                sh './gradlew build -x test'
+            }
+        }
+       stage('Test') {
+            steps {
+                sh './gradlew check'
+            }
+            post {
+              always {
+                junit 'build/reports/tests/*.xml'
```
Comment (timja) [inline Jenkinsfile:18]:
```suggestion
                // enable when tests are added junit 'build/reports/tests/*.xml'
```

**Comment by NotMyFault** [comment]
> Application doesn't even start though with your recent changes 👅.

I was not _quite_ there yet, but them being compile time dependencies makes not much sense oops 🤦🏻

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
+pipeline {
+    agent {
+        label 'jdk11'
+    }
+
+    stages {
+        stage('Build') {
+            steps {
+                sh './gradlew build -x test'
```
Comment (halkeye) [inline Jenkinsfile:9]: test "works" fine when i have implementation vs compile only, though i'm not sure the test is really testing anything

Comment (timja) [reply Jenkinsfile:9]: it's not but generally you would separate these out.

test failed for me with compileOnly

Comment (halkeye) [reply Jenkinsfile:9]: oh you are saying run tests as a different step. Thats cool.

I think the tests work fine once moved to implementation

Comment (timja) [reply Jenkinsfile:9]: they do yes

**Review by dduportal — APPROVED** [review]
LGTM 👍

**Pull request merged**